featurebyte/core/accessor/vector.py

```diff
@@ -0,0 +1,70 @@
+"""
+Vector accessor module
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, TypeVar
+
+from featurebyte.common.doc_util import FBAutoDoc
+from featurebyte.core.util import series_binary_operation
+from featurebyte.enum import DBVarType
+from featurebyte.query_graph.enum import NodeType
+
+if TYPE_CHECKING:
+    from featurebyte.core.series import FrozenSeries
+else:
+    FrozenSeries = TypeVar("FrozenSeries")
+
+
+class VectorAccessorMixin:
+    """
+    VectorAccessorMixin class
+    """
+
+    @property
+    def vec(self: FrozenSeries) -> VectorAccessor:  # type: ignore
+        """
+        Vector accessor object
+
+        Returns
+        -------
+        VectorAccessor
+        """
+        return VectorAccessor(self)
+
+
+class VectorAccessor:
+    """
+    VectorAccessor class used to manipulate vector type FrozenSeries object
+    """
+
+    # documentation metadata
+    __fbautodoc__ = FBAutoDoc(
+        proxy_class="featurebyte.VectorAccessor",
+    )
+
+    def __init__(self, obj: FrozenSeries):
+        if obj.dtype != DBVarType.ARRAY:
+            raise AttributeError("Can only use .vec accessor with ARRAY values!")
+        self._obj = obj
+
+    def cosine_similarity(self, other: FrozenSeries) -> FrozenSeries:
+        """
+        Calculate cosine similarity between two vector series.
+
+        Parameters
+        ----------
+        other: FrozenSeries
+            The other vector series
+
+        Returns
+        -------
+        FrozenSeries
+        """
+        return series_binary_operation(
```

Comment:
Should we validate that the other series is also of vector type? Is it possible to validate that the vectors are of the same dimension?

Reply:
hm yea i'll add the validation!

do they have to be of the same dimension for cosine similarity? i'm not too familiar w/ how to interpret the values, but the current UDFs will support vectors of different dimensions.

Reply:
The current implementation implicitly truncates the extra dimensions in the longer vector. I'm not sure what is the desired behaviour here, might want to clarify on that.

Reply:
i'll merge this in first and follow-up with changes if necessary!